Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -293,7 +311,7 @@ void GraphicsContextSkia::drawLine(const FloatPoint& point1, const FloatPoint& p
         return;
 
     SkPaint paint = createStrokePaint();
-    paint.setColor(SkColor(strokeColor().colorWithAlphaMultipliedBy(alpha())));
+    paint.setColor(strokeColor().colorWithAlphaMultipliedBy(alpha()), colorSpace().platformColorSpace().get());
```

Comment:
I still think the color space parameters means the color space of the given color, not the destination color space. When converting Color to Skcolor we always get the Color as SRGBA, so I don't think we need to pass any color space to SkPaint::setColor

Reply:
You're correct color space parameters mean the color space of the given color. Also, as you mentioned - when converting Color to SkColor we always get the Color as SRGBA. The thing is, the callers of GraphicsContext assume that the color they provide will be exactly what will be written to the destination. So if some caller (let's say feFlood filter) uses GraphicsContext that has linear SRGB destination and calls fillRect with color `(0.501961, 0.501961, 0.501961, 1.000000)` it assumes the pixels will have values `(0.501961, 0.501961, 0.501961, 1.000000)`. Without the above parameter, the color would be treated as SRGB and therefore while filling rect it would be converted to linear SRGB so the pixels would be filled with color `(0.215686, 0.215686, 0.215686, 1.000000)`

Reply:
Actually, I've started checking callers on why they _"assume that the color they provide should be exactly what will be written to the destination"_ and I've discovered that they actually don't assume that and the problem is elsewhere - I'll update PR in few h.

Reply:
Done, see https://github.com/WebKit/WebKit/pull/27079#discussion_r1565786804